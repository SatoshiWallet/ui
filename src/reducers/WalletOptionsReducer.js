// @flow

import { type Reducer, combineReducers } from 'redux'

import type { Action } from '../types/reduxTypes.js'
import { xPubExplorer, xPubSyntax } from './XPubModalReducer.js'

export type WalletListState = {
  +viewXPubWalletModalVisible: boolean,
  +walletArchivesVisible: boolean,
  +walletId: string,
  +xPubSyntax: string,
  +xPubExplorer: string
}

const walletId = (state = '', action: Action): string => {
  switch (action.type) {
    case 'OPEN_VIEWXPUB_WALLET_MODAL': {
      if (!action.data) throw new Error('Invalid action')
      return action.data.walletId
    }

    case 'CLOSE_VIEWXPUB_WALLET_MODAL': {
      return ''
    }

    default:
      return state
  }
}

const walletArchivesVisible = (state = false, action: Action): boolean => {
  switch (action.type) {
    // case ACTION.OPEN_WALLET_ARCHIVES:
    //   return true
    // case ACTION.CLOSE_WALLET_ARCHIVES:
    //   return false
    default:
      return state
  }
}

const viewXPubWalletModalVisible = (state = false, action: Action): boolean => {
  switch (action.type) {
    case 'OPEN_VIEWXPUB_WALLET_MODAL': {
      return true
    }

    case 'CLOSE_VIEWXPUB_WALLET_MODAL': {
      return false
    }

    default:
      return state
  }
}

export const walletList: Reducer<WalletListState, Action> = combineReducers({
  viewXPubWalletModalVisible,
  walletArchivesVisible,
  walletId,
  xPubSyntax,
  xPubExplorer
})
