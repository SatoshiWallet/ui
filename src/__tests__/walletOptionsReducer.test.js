// @flow

/* globals describe it expect */

import { walletList as reducer } from '../reducers/WalletOptionsReducer.js'

const dummyAction = { type: 'DUMMY_ACTION_PLEASE_IGNORE' }

describe('reducer', () => {
  const initialState = reducer(undefined, dummyAction)

  it('initial state', () => {
    const actual = reducer(undefined, dummyAction)

    expect(actual).toMatchSnapshot()
  })

  it('open viewxpub wallet modal', () => {
    const action = {
      type: 'OPEN_VIEWXPUB_WALLET_MODAL',
      data: { walletId: 'VIEWXPUB WALLET ID', xPub: 'VIEWXPUB xpub1892736451987263r', xPubExplorer: '' }
    }
    const actual = reducer(initialState, action)

    expect(actual).toMatchSnapshot()
  })
})
