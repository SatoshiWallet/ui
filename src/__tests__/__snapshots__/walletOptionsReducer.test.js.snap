// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer initial state 1`] = `
Object {
  "viewXPubWalletModalVisible": false,
  "walletArchivesVisible": false,
  "walletId": "",
  "xPubExplorer": "",
  "xPubSyntax": "",
}
`;

exports[`reducer open viewxpub wallet modal 1`] = `
Object {
  "viewXPubWalletModalVisible": true,
  "walletArchivesVisible": false,
  "walletId": "VIEWXPUB WALLET ID",
  "xPubExplorer": "",
  "xPubSyntax": "VIEWXPUB xpub1892736451987263r",
}
`;
